File: schemaMapper/mappingEngine/src/main/java/de/iisys/sysint/hicumes/mappingEngine/models/MappingAndDataSource.java
package de.iisys.sysint.hicumes.mappingEngine.models;

import de.iisys.sysint.hicumes.mappingEngine.models.dataSource.reader.DataReader;
import de.iisys.sysint.hicumes.mappingEngine.models.dataSource.reader.ReaderResult;
import de.iisys.sysint.hicumes.mappingEngine.models.dataSource.writer.DataWriter;
import de.iisys.sysint.hicumes.mappingEngine.models.mapping.MappingConfiguration;
import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Data
@NamedQueries({
        @NamedQuery(name = "MappingAndDataSource.findAll", query = "SELECT t FROM MappingAndDataSource t")
        , @NamedQuery(name = "MappingAndDataSource.findById", query = "SELECT t FROM MappingAndDataSource t WHERE t.id = :id")
})
public class MappingAndDataSource {
    public MappingAndDataSource(@NotNull String name, @NotNull DataReader dataReader, @NotNull DataWriter dataWriter, @NotNull MappingConfiguration mappingConfiguration, @NotNull ReaderResult readerResult) {
        this.name = name;
        this.dataReader = dataReader;
        this.dataWriter = dataWriter;
        this.mappingConfiguration = mappingConfiguration;
        this.readerResult = readerResult;
    }

    public MappingAndDataSource() {
    }

    @Id
    protected long id;

    protected String externalId;

    @NotNull
    private String name;

    @NotNull
    @OneToOne(orphanRemoval = true, cascade = {CascadeType.REMOVE, CascadeType.MERGE, CascadeType.PERSIST})
    private DataReader dataReader;

    @NotNull
    @OneToOne(orphanRemoval = true, cascade = {CascadeType.REMOVE, CascadeType.MERGE, CascadeType.PERSIST})
    private DataWriter dataWriter;

    @NotNull
    @OneToOne(orphanRemoval = true, cascade = {CascadeType.REMOVE, CascadeType.MERGE, CascadeType.PERSIST})
    private MappingConfiguration mappingConfiguration;

    @NotNull
    @OneToOne(orphanRemoval = true, cascade = {CascadeType.REMOVE, CascadeType.MERGE, CascadeType.PERSIST})
    private ReaderResult readerResult;
}

File: docker_compose/feinplanung/HiCuMes.Persistence/Entities/Mappinganddatasource.cs
using System;
using System.Collections.Generic;

namespace HiCuMes.Persistence.Entities;

public partial class Mappinganddatasource
{
    public long Id { get; set; }

    public string? ExternalId { get; set; }

    public string Name { get; set; } = null!;

    public long DataReaderId { get; set; }

    public long DataWriterId { get; set; }

    public long MappingConfigurationId { get; set; }

    public long ReaderResultId { get; set; }

    public  Datareader DataReader { get; set; } = null!;

    public  Datawriter DataWriter { get; set; } = null!;

    public  Mappingconfiguration MappingConfiguration { get; set; } = null!;

    public  Readerresult ReaderResult { get; set; } = null!;
}
